manager/state/raft/raft.go

```diff
@@ -1703,15 +1707,27 @@ func (n *Node) processInternalRaftRequest(ctx context.Context, r *api.InternalRa
 	select {
 	case x, ok := <-ch:
 		if !ok {
-			return nil, ErrLostLeadership
+			if err, ok := x.(error); ok {
```

Comment:
If `ok` is false it means the channel was closed and nothing was received, so trying to use `x` is wrong. Also, errors are never sent over this channel.

Reply:
Its orca#9393. 

>If ok is false it means the channel was closed and nothing was received, so trying to use x is wrong. Also, errors are never sent over this channel.

Thanks for pointing this out!
I guess I need to re-read the code. The reason I'm doing this is because I think assuming ErrLostLeadership here may not accurate (please correct me I'm wrong). So, I'm trying to see if a more meaningful error can be returned.

Reply:
I believe it is accurate. That's the only thing that causes proposals to get canceled.

Reply:
This is in investigating a case where proposal fails with ErrLostLeadership but the leader does not actually lose leadership. Unfortunately, its not reproducible any more.